lib/tmuxinator/cli.rb

```diff
@@ -226,6 +227,44 @@ def show_continuation_prompt
       def kill_project(project)
         Kernel.exec(project.kill)
       end
+
+      def tmux_session_exists
+        ENV["TMUX"] ? true : false
+      end
+
+      def should_append?(options)
+        options[:append] == true ||
+          (options[:append] != false && Tmuxinator::Config.options[:append])
+      end
+
+      def append_in_session(options, name = nil)
+        has_tmux_session = nil
+        if should_append?(options)
+          has_tmux_session = tmux_session_exists
+          if !has_tmux_session
+            say "Creating session '#{name}'"
+          end
+        end
+        has_tmux_session
+      end
```

Comment:
This method is somewhat hard to follow because it's responsible for two things: (a) deciding whether a new session is created *and* (b) printing a message with the new session name. Given (a), it should be named `append_in_session?` or something like that, and always return a Boolean (right now it returns `true`, `false` or `nil`—the rest of the code doesn't appear to distinguish between the `false` and `nil` possibilities). But since it also has a side effect of printing a message and takes this extra `name` parameter, that wouldn't be right either. These things don't seem to belong in the same method.

In any case, it could be simplified to:

```suggestion
      def append_in_session(options, name = nil)
        if should_append?(options) && !tmux_session_exists
          say "Creating session '#{name}'"
          true
        end
      end
```